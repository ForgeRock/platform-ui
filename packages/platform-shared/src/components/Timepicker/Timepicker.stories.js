/**
 * Copyright 2020 ForgeRock AS. All Rights Reserved
 *
 * Use of this code requires a commercial software license with ForgeRock AS.
 * or with one of its affiliates. All use shall be exclusively subject
 * to such license between the licensee and ForgeRock AS.
 */
/* eslint import/no-extraneous-dependencies: 0 */
import { storiesOf } from '@storybook/vue';
import { withKnobs, text } from '@storybook/addon-knobs';
import {
  BCard,
  BContainer,
} from 'bootstrap-vue';
import Timepicker from './index';

const stories = storiesOf('Components|Timepicker', module).addParameters({ component: Timepicker });

stories.addDecorator(withKnobs);

const template = `
  <div class="margin-25">
    <div class="sg_masthead">
      <BContainer>
        <h1 class="display-4">Timepicker</h1>
        <p class="mb-4 lead">
          Timepicker is a wrapper for bootstrap's Timepicker component.
          Adds a floating label and custom icons.
        </p>
      </BContainer>
    </div>
    <BContainer>
      <section class="sg_pattern">
        <h2>Basic Example</h2>
        <p class="mb-5">
          Select a time
        </p>
        <BCard>
          <FrTimepicker 
            :placeholder="placeholder"
            v-model="value" />
          Value: {{ value }}
        </BCard>
      </section>
    </BContainer>
    <BContainer>
      <section class="sg_pattern">
        <h2>Attribute passthrough</h2>
        <p class="mb-5">
          All props and events are passed through to bootstrap component. This example uses minutes-step="15" and size="lg"
        </p>
        <BCard>
          <FrTimepicker 
            minutes-step="15"
            size="lg"
            :placeholder="placeholder"
            v-model="value" />
          Value: {{ value }}
        </BCard>
      </section>
    </BContainer>
  </div>
`;

stories.add('Timepicker', () => ({
  components: {
    FrTimepicker: Timepicker,
    BCard,
    BContainer,
  },
  template,
  data() {
    return {
      value: '',
    };
  },
  props: {
    placeholder: {
      type: String,
      default: text('placeholder', 'Placeholder Text'),
    },
  },
}));
