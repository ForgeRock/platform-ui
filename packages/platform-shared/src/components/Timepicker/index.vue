<!-- Copyright 2020 ForgeRock AS. All Rights Reserved

Use of this code requires a commercial software license with ForgeRock AS.
or with one of its affiliates. All use shall be exclusively subject
to such license between the licensee and ForgeRock AS. -->
<template>
  <BFormTimepicker
    v-bind="$attrs"
    v-on="$listeners"
    menu-class="shadow-lg"
    no-close-button
    :class="{'input-has-value': value}"
    :label-selected="placeholder"
    :placeholder="placeholder"
    :value="value">
    <template #button-content>
      <i class="material-icons-outlined">
        access_time
      </i>
    </template>
  </BFormTimepicker>
</template>

<script>
import {
  BFormTimepicker,
} from 'bootstrap-vue';

/**
 * Bootstrap timepicker with custom icons, default values set.
 * v-model value returns 'hh:mm:ss'.
 */
export default {
  name: 'Timepicker',
  components: {
    BFormTimepicker,
  },
  props: {
    /**
     * v-model value
     */
    value: {
      type: String,
      default: '',
    },
    /**
     * Text label. Floats when a value is selected.
     */
    placeholder: {
      type: String,
      default: '',
    },
  },
};
</script>
<style lang="scss" scoped>
// floating label support
.input-has-value {
  /deep/ label {
    padding-top: 1.25rem;
    padding-bottom: 0.25rem;
    position: relative;

    .sr-only {
      width: auto;
      height: auto;
      padding-top: 0.25rem;
      padding-bottom: 0.25rem;
      margin: 0;
      overflow: hidden;
      clip: auto;
      top: 0;
      left: 0.25rem;
      font-size: 12px;
      color: $gray-600;
    }
  }
}

.dropdown {
  padding: 0;
}
</style>
