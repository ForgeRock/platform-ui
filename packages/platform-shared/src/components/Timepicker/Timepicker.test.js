/**
 * Copyright 2020 ForgeRock AS. All Rights Reserved
 *
 * Use of this code requires a commercial software license with ForgeRock AS.
 * or with one of its affiliates. All use shall be exclusively subject
 * to such license between the licensee and ForgeRock AS.
 */
import { shallowMount } from '@vue/test-utils';
import Timepicker from './index';

describe('Timepicker Component', () => {
  it('Timepicker successfully loaded', () => {
    const wrapper = shallowMount(Timepicker);
    expect(wrapper.name()).toEqual('Timepicker');
  });

  it('contains a bootstrap timepicker', () => {
    const wrapper = shallowMount(Timepicker);
    expect(wrapper.find({ name: 'BFormTimepicker' }).exists()).toBe(true);
  });

  it('has placeholder text', () => {
    const wrapper = shallowMount(Timepicker, {
      propsData: {
        placeholder: 'test',
      },
    });
    expect(wrapper.find({ name: 'BFormTimepicker' }).attributes('placeholder')).toBe('test');
  });

  it('floats placeholder text when a value is set', () => {
    const wrapper = shallowMount(Timepicker, {
      propsData: {
        placeholder: 'test',
      },
    });
    expect(wrapper.find('.input-has-value').exists()).toBe(false);
    wrapper.setProps({ value: 'John' });
    expect(wrapper.find('.input-has-value').exists()).toBe(true);
  });
});
