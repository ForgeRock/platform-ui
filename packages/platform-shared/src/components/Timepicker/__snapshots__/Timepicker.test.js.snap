// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timepicker Component contains a bootstrap timepicker: console-warn-snapshot 1`] = `
[
  [
    "[Vue warn]: (deprecation RENDER_FUNCTION) Vue 3's render function API has changed. You can opt-in to the new API with:

  configureCompat({ RENDER_FUNCTION: false })

  (This can also be done per-component via the "compatConfig" option.)
  Details: https://v3-migration.vuejs.org/breaking-changes/render-function-api.html",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation INSTANCE_LISTENERS) vm.$listeners has been removed. In Vue 3, parent v-on listeners are included in vm.$attrs and it is no longer necessary to separately use v-on="$listeners" if you are already using v-bind="$attrs". (Note: the Vue 3 behavior only applies if this compat config is disabled)
  Details: https://v3-migration.vuejs.org/breaking-changes/listeners-removed.html",
    "
",
    " at <Timepicker",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation GLOBAL_EXTEND) (2)",
    "
",
    " at <Timepicker",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "buttonOnly" with v-bind value \`false\` will render buttonOnly="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BFormTimepicker",
    "menu-class="shadow-lg"",
    "no-close-button=""",
    "class=""",
    " ...",
    ">",
    "
",
    " at <Timepicker",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "disabled" with v-bind value \`false\` will render disabled="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BFormTimepicker",
    "menu-class="shadow-lg"",
    "no-close-button=""",
    "class=""",
    " ...",
    ">",
    "
",
    " at <Timepicker",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "dropleft" with v-bind value \`false\` will render dropleft="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BFormTimepicker",
    "menu-class="shadow-lg"",
    "no-close-button=""",
    "class=""",
    " ...",
    ">",
    "
",
    " at <Timepicker",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "dropright" with v-bind value \`false\` will render dropright="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BFormTimepicker",
    "menu-class="shadow-lg"",
    "no-close-button=""",
    "class=""",
    " ...",
    ">",
    "
",
    " at <Timepicker",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "dropup" with v-bind value \`false\` will render dropup="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BFormTimepicker",
    "menu-class="shadow-lg"",
    "no-close-button=""",
    "class=""",
    " ...",
    ">",
    "
",
    " at <Timepicker",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "hideHeader" with v-bind value \`false\` will render hideHeader="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BFormTimepicker",
    "menu-class="shadow-lg"",
    "no-close-button=""",
    "class=""",
    " ...",
    ">",
    "
",
    " at <Timepicker",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "noFlip" with v-bind value \`false\` will render noFlip="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BFormTimepicker",
    "menu-class="shadow-lg"",
    "no-close-button=""",
    "class=""",
    " ...",
    ">",
    "
",
    " at <Timepicker",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "nowButton" with v-bind value \`false\` will render nowButton="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BFormTimepicker",
    "menu-class="shadow-lg"",
    "no-close-button=""",
    "class=""",
    " ...",
    ">",
    "
",
    " at <Timepicker",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "required" with v-bind value \`false\` will render required="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BFormTimepicker",
    "menu-class="shadow-lg"",
    "no-close-button=""",
    "class=""",
    " ...",
    ">",
    "
",
    " at <Timepicker",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "resetButton" with v-bind value \`false\` will render resetButton="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BFormTimepicker",
    "menu-class="shadow-lg"",
    "no-close-button=""",
    "class=""",
    " ...",
    ">",
    "
",
    " at <Timepicker",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "right" with v-bind value \`false\` will render right="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BFormTimepicker",
    "menu-class="shadow-lg"",
    "no-close-button=""",
    "class=""",
    " ...",
    ">",
    "
",
    " at <Timepicker",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "showSeconds" with v-bind value \`false\` will render showSeconds="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BFormTimepicker",
    "menu-class="shadow-lg"",
    "no-close-button=""",
    "class=""",
    " ...",
    ">",
    "
",
    " at <Timepicker",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
]
`;
